encord/http/bundle.py

```diff
@@ -0,0 +1,106 @@
+import logging
+from typing import Callable, TypeVar, Optional, List, Dict, Tuple, Generic, Generator
+from dataclasses import asdict, dataclass, field
+from functools import reduce
+
+log = logging.getLogger(__name__)
+
+T = TypeVar("T")
+R = TypeVar("R")
+
+
+@dataclass
+class BundleResultHandler(Generic[T]):
+    predicate: str
+    continuation: Callable[[T], None]
+
+
+@dataclass
+class BundledOperation(Generic[T, R]):
+    operation: Callable[..., List[R]]
+    reducer: Callable[[T, T], T]
+    mapper: Optional[Callable[[R], str]]
+    limit: int
+    payloads: list = field(default_factory=lambda: [])
```

Comment:
I'm curious, is this always a "list of lists"? The usage right now suggests so. If yes, maybe we can be more precise with the type here?